src/lightning/pytorch/callbacks/spike.py

```diff
@@ -0,0 +1,31 @@
+import os
+from typing import Any, Mapping, Union
+
+import torch
+
+import lightning.pytorch as pl
+from lightning.fabric.utilities.spike import SpikeDetection as FabricSpikeDetection
+from lightning.pytorch.callbacks.callback import Callback
+
+
+class SpikeDetection(FabricSpikeDetection, Callback):
+    @torch.no_grad()
+    def on_train_batch_end(  # type: ignore
+        self,
+        trainer: pl.Trainer,
+        pl_module: pl.LightningModule,
+        outputs: Union[torch.Tensor, Mapping[str, torch.Tensor]],
+        batch: Any,
+        batch_idx: int,
+    ) -> None:
+        if isinstance(outputs, torch.Tensor):
+            loss = outputs.detach()
+        elif isinstance(outputs, Mapping):
+            loss = outputs["loss"].detach()
```

Comment:
Maybe nicer error if `loss` key is missing

Reply:
from a training_step it can't be missing. Otherwise optimization (specifically backward) would fail earlier.